src/Main.elm

```diff
@@ -41,25 +147,151 @@ update msg model =
+            ]
+        , div [ class "buttons" ]
+            [ button
+                [ class buttonClass
+                , onClick clickEvent
+                ]
+                [ text buttonText
+                ]
+            , button [ onClick (Decrement book.bookID) ] [ text "-" ]
+            , text (String.fromInt (getQuantity cart book.bookID))
+            , button [ onClick (Increment book.bookID) ] [ text "+" ]
             ]
         ]
 
 
 
+-- cartView : Model -> Html Msg
+-- cartView model =
+--     div []
+--         [ div []
+--             [ if List.isEmpty model.cart then
+--                 div [ class "remove" ]
+--                     [ text "Cart is empty" ]
+--
+--               else
+--                 div [] (List.map (\c -> cartItemView c model.books) model.cart)
+--             ]
+--         , div [] [ text <| "Subtotal: $" ++ String.fromInt (cartTotalView model) ]
+--         ]
+
+
+cartView : Model -> Html Msg
+cartView model =
+    div []
+        [ table []
+            ([ thead []
+                [ th [] [ text "Item" ]
+                , th [] [ text "Quantity" ]
+                , th [] [ text "Amount" ]
+                ]
+             ]
+                ++ List.map (\c -> cartItemView c model.books) model.cart
+                ++ [ tr
+                        []
+                        [ td [] [ text "Total" ]
+                        , td [] []
+                        , td [] [ text <| "$" ++ String.fromInt (cartTotalView model) ]
+                        ]
+                   ]
+            )
+        ]
+
+
+
+-- cartItemView : CartItem -> List Book -> Html Msg
+-- cartItemView cartItem bookList =
+--     div []
+--         [ div [] [ text <| getName cartItem.id bookList ++ " x" ++ String.fromInt cartItem.quantity ++ " " ++ "$" ++ String.fromInt (getPrice cartItem.id cartItem.quantity bookList) ]
+--         ]
+
+
+cartItemView : CartItem -> List Book -> Html Msg
+cartItemView cartItem bookList =
+    tr []
+        [ td [] [ text <| getName cartItem.id bookList ]
+        , td [] [ text (String.fromInt cartItem.quantity) ]
+        , td [] [ text <| "$" ++ String.fromInt (getPrice cartItem.id cartItem.quantity bookList) ]
+        ]
+
+
+getName : BookID -> List Book -> String
+getName bookID bookList =
+    let
+        result =
+            List.filter (\b -> b.bookID == bookID) bookList
+    in
+    case List.head result of
+        Nothing ->
+            ""
+
+        Just b ->
+            b.title
+
+
+getPrice : BookID -> Int -> List Book -> Int
+getPrice bookID quant bookList =
+    let
+        result =
+            List.filter (\b -> b.bookID == bookID) bookList
+    in
+    case List.head result of
+        Nothing ->
+            0
+
+        Just b ->
+            b.price * quant
+
+
+cartTotalView : Model -> Int
+cartTotalView model =
+    List.sum (List.map (\item -> getPrice item.id item.quantity model.books) model.cart)
```

Comment:
Good case for a `reduce` FN

Reply:
Like a fold? I tried writing this as

`List.foldr (\item -> getPrice item.id item.quantity model.books) 0 model.cart`

which seems like it should work, but got the slightly opaque error message

"This argument is an anonymous function of type:

    { a | id : BookID, quantity : Int } -> Int

But `foldr` needs the 1st argument to be:

    { a | id : BookID, quantity : Int } -> b "

- why can't Int be, well, b?